spec/features/rental_creation_spec.rb

```diff
@@ -0,0 +1,103 @@
+require 'rails_helper'
+
+describe 'Creating a new rental', js: true do
+
+  before(:each) do
+    # In the current state of this, these tests are using development data, so
+    # it's acting as if the @current_user delegation is based on development
+    # logic. That is, it is using the first user available in the development
+    # database.
+
+    @tomorrow = Date.today + 1
+    @day_after = @tomorrow + 1
+  end
+
+  describe 'static page tests for the sake of testing' do
+
+  end
+
+  describe 'form autopopulation' do
+    before(:each) do
+      visit '/rentals/new'
+      @today = Date.today.strftime('%Y-%m-%d')
+    end
+
+    # removed data_type_id test due to the weirdness of the generated select
+    # object using 'data-id' instead of 'id' so find_by_id does not work.
+
+    it 'populates fields' do
+      # Normally I would spread this to 3 tests, but due to the way integration
+      # tests are ran (using the browser), it is a significant performance gain
+      # to place all 3 of these in 1 test.
+
+      expect(find_by_id('rental_start_time').value).to eql @today
+      expect(find_by_id('rental_end_time').value).to eql @today
+      expect(find_by_id('rental_date_range').value).to eql '1'
+    end
+  end
+
+  describe 'rental creation' do
+    describe 'invalid' do
+      it 'redirects to creation page again' do
+        visit '/rentals/new'
+
+        within('form.new_rental') do
+          fill_in('rental_start_time', with: @tomorrow.strftime('%Y-%m-%d'))
+          fill_in('rental_end_time', with: Date.today.strftime('%Y-%m-%d'))
```

Comment:
you already assigned `Date.today.strftime('%Y-%m-%d')` to `@today`, might as well use it